L1TriggerConfig/Utilities/src/L1TMuonBarrelKalmanParamsViewer.cc

```diff
@@ -0,0 +1,77 @@
+#include <iomanip>
+#include <iostream>
+
+#include "FWCore/Framework/interface/EDAnalyzer.h"
+
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/Framework/interface/ESHandle.h"
+
+#include "CondFormats/DataRecord/interface/L1TMuonBarrelKalmanParamsRcd.h"
+#include "CondFormats/L1TObjects/interface/L1TMuonBarrelKalmanParams.h"
+#include "FWCore/ServiceRegistry/interface/Service.h"
+#include "CondCore/DBOutputService/interface/PoolDBOutputService.h"
+#include "CondCore/CondDB/interface/Session.h"
+
+#include <iostream>
+using namespace std;
+
+class L1TMuonBarrelKalmanParamsViewer : public edm::EDAnalyzer {
+private:
+  std::string hash(void *buf, size_t len) const;
+
+public:
+  void analyze(const edm::Event &, const edm::EventSetup &) override;
+
+  L1TMuonBarrelKalmanParamsViewer(const edm::ParameterSet &){};
+  ~L1TMuonBarrelKalmanParamsViewer(void) override {}
+};
+
+#include <openssl/sha.h>
+#include <cmath>
+#include <iostream>
+using namespace std;
+
+std::string L1TMuonBarrelKalmanParamsViewer::hash(void *buf, size_t len) const {
+  char tmp[SHA_DIGEST_LENGTH * 2 + 1];
+  bzero(tmp, sizeof(tmp));
+  SHA_CTX ctx;
+  if (!SHA1_Init(&ctx))
+    throw cms::Exception("L1TMuonBarrelKalmanParamsViewer::hash") << "SHA1 initialization error";
+
+  if (!SHA1_Update(&ctx, buf, len))
+    throw cms::Exception("L1TMuonBarrelKalmanParamsViewer::hash") << "SHA1 processing error";
+
+  unsigned char hash[SHA_DIGEST_LENGTH];
+  if (!SHA1_Final(hash, &ctx))
+    throw cms::Exception("L1TMuonBarrelKalmanParamsViewer::hash") << "SHA1 finalization error";
+
+  // re-write bytes in hex
+  for (unsigned int i = 0; i < 20; i++)
+    ::sprintf(&tmp[i * 2], "%02x", hash[i]);
+
+  tmp[20 * 2] = 0;
+  return std::string(tmp);
+}
+
+void L1TMuonBarrelKalmanParamsViewer::analyze(const edm::Event &iEvent, const edm::EventSetup &evSetup) {
+  edm::ESHandle<L1TMuonBarrelKalmanParams> handle1;
+  evSetup.get<L1TMuonBarrelKalmanParamsRcd>().get(handle1);
+  std::shared_ptr<L1TMuonBarrelKalmanParams> ptr(new L1TMuonBarrelKalmanParams(*(handle1.product())));
+
+  // Get the nodes and print out
+  auto pnodes = ptr->pnodes_[ptr->CONFIG];
+  cout << "version    : " << ptr->version_ << endl;
+  cout << "fwVersion  : " << hex << pnodes.fwVersion_ << dec << endl;
+  cout << "LUTsPath   : " << pnodes.kalmanLUTsPath_ << endl;
```

Comment:
std::cout is not threadsafe, please use https://twiki.cern.ch/twiki/bin/view/CMSPublic/SWGuideMessageLogger

Reply:
Hi, thanks for the notice about this
This module here is an L1T private tester and is not used in any central CMSSW workflow.
The purpose is to read 1 event (only) and print out the info included in the db Rcd
However, if it's required we can update it with MesageLogger in the next iteration of this development, is this fine?

Reply:
@panoskatsoulis  For the next iteration: if it is a private test, it shoud be better moved into the \test area?

Reply:
Hi @perrotta, this is a good point
it is the entire pkg `L1TriggerConfig/Utilities` that contains these testing modules
Let me discuss this with the other L1 guys tho, thx